.github/workflows/cifuzz.yml

```diff
@@ -0,0 +1,26 @@
+name: CIFuzz
+on: [pull_request]
+jobs:
+  Fuzzing:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Build Fuzzers
+      id: build
+      uses: google/oss-fuzz/infra/cifuzz/actions/build_fuzzers@master
```

Comment:
We generally prefer to have actions pinned to specific SHAs in order to avoid "invisible" changes in our CI, in particular for security reasons. The SHAs are them bumped regularly by Dependabot in a new PR when new action releases come out, which offers the chance to review the changes.

It seems the `google/oss-fuzz` project does not provide any kind of release tracking for its actions, hence if we pin to a SHA it won't be tracked by Dependabot. Is that correct?

Reply:
I am happy to fix it in on a SHA, although this may cause issues down the line https://github.com/linkerd/linkerd2-proxy/pull/1336#discussion_r758701279

I think you're right about the tracking, @jonathanmetzman could you confirm this?

In essence, the SHA will have to be updated in case the set up (`Dockerfile` and `build.sh`) in OSS-Fuzz changes. This year, a few changes (~3) has happened in CIlium https://github.com/google/oss-fuzz/commits/master/projects/cilium 

Do you still prefer to pin to a SHA?

Reply:
Right we don't really do releases.

Reply:
The issue I see with not pinning is that the fuzzing job might change in the middle of multiple runs in the same PR. Security is less of a concern for this workflow because the fuzzing is done by OSS-Fuzz and thus we can have this workflow run on `pull_request`, which means it won't have access to our secrets when run from a fork (as in this PR).

We can either decide to pin and manually bump the SHAs once in a while, as we do with some excluded dependabot dependencies, or accept that we point to `master` for the ease of use.

/cc @aanm @joestringer for more opinions

Reply:
I'm fine with pinning to a sha+periodically bumping it manually, with the idea that hopefully long term dependabot can take care of that for us. I assume that if we're paying attention to this workflow and it's providing value, then we will be motivated to keep it up to date.

Reply:
> The issue I see with not pinning is that the fuzzing job might change in the middle of multiple runs in the same PR.

Yes, this is the main point. @DavidKorczynski if I may suggest it would be best to have releases, that way dependabot can to the upgrades automatically for us.